pilot/pkg/xds/xdsgen.go

```diff
@@ -187,3 +188,45 @@ func ResourceSize(r model.Resources) int {
 	}
 	return size
 }
+
+// xdsNeedsPush checks for the common cases whether we need to push or not.
+func xdsNeedsPush(req *model.PushRequest, proxy *model.Proxy, allowIncrementalPush bool) (bool, bool) {
```

Comment:
make return value *bool ？

Reply:
don't prefer that. But if that is preferred, no strong objection to change

Reply:
Or make the named return value, so we can distinguish the two bool meaning